src/tracer_factory.h

```diff
@@ -19,28 +19,9 @@ ot::expected<TracerOptions> optionsFromConfig(const char *configuration,
 template <class TracerImpl>
 class TracerFactory : public ot::TracerFactory {
  public:
-  // Accepts configuration in JSON format, with the following keys:
-  // "service": Required. A string, the name of the service.
-  // "agent_host": A string, defaults to localhost. Can also be set by the environment variable
-  //     DD_AGENT_HOST
-  // "agent_port": A number, defaults to 8126. "type": A string, defaults to web. Can also be set
-  //     by the environment variable DD_TRACE_AGENT_PORT
-  // "type": A string, defaults to web.
-  // "environment": A string, defaults to "". The environment this trace belongs to.
-  //     eg. "" (env:none), "staging", "prod". Can also be set by the environment variable
-  //     DD_ENV
-  // "sample_rate": A double, defaults to 1.0.
-  // "operation_name_override": A string, if not empty it overrides the operation name (and the
-  //     overridden operation name is recorded in the tag "operation").
-  // "propagation_style_extract": A list of strings, each string is one of "Datadog", "B3".
-  //     Defaults to ["Datadog"]. The type of headers to use to propagate
-  //     distributed traces. Can also be set by the environment variable
-  //     DD_PROPAGATION_STYLE_EXTRACT.
-  // "propagation_style_inject": A list of strings, each string is one of "Datadog", "B3". Defaults
-  //     to ["Datadog"]. The type of headers to use to receive distributed traces. Can also be set
-  //     by the environment variable DD_PROPAGATION_STYLE_INJECT.
-  //
-  // Extra keys will be ignored.
+  // Accepts configuration as a JSON object.  See `optionsFromConfig` in
```

Comment:
I chose the latter because I didn't feel like writing the documentation myself, and saw that it might fall out of sync anyway :)

I couldn't find a place in the public documentation, either, that describes everything supported here. One thing to consider is making this a public contract, maintenance and all, and then linking to it from the public docs.

Not today, though.